### ISSUE TITLE
selector-anb-no-unmatchable false alarms

### ISSUE THREAD

**Issue by woldtwerk** [stylelint/stylelint#6719]
### What steps are needed to reproduce the bug?

error first line, first character
```css
@layer tailwind-components {
  .foo {
     display: block;
  }
}
```

error first line, first character
```css
:root {
  --c-foo: red;
}

```

worst part is that settings:

```json
"rules": {
      "selector-anb-no-unmatchable": null
    }
```
is completely ignored.

### What Stylelint configuration is needed to reproduce the bug?

```js
extends: [
    'stylelint-config-standard',
  ],
```

### How did you run Stylelint?

pnpm exec stylelint \"web/themes/custom/**/*.css\" --allow-empty-input --ignore-path=.gitignore --fix

### Which version of Stylelint are you using?

15.3.0

### What did you expect to happen?

no error

being able to disasble the rule

### What actually happened?

Unknown rule selector-anb-no-unmatchable selector-anb-no-unmatchable

**Comment by ybiquitous** [comment]
@woldtwerk Thanks for the report using the template. I tried reproducing the error, but I couldn't (see the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAbAhgTxgJwAQBc0BLFAdyIDsoBaSAWwAcIKYL8BnXYAHQu-wB0AMwgQuvfvyhF2DdBni4ARighgA1gG4J+AL699fCvGyj84vvmq1qIiIuwwo2ioZAAaEEJIwAcmjo4RBgADwC5GAEwdnYPcGZvAHMEEB4KXFxuEFD8Vih2LMUs9nwMFBgUSitICiTqErQqNGwoLIM4mqSAMQhsOjR8FIArdmY42AZYxDSM4tLyyrZCzJAARgBWAQBmAQAGLPdeWZASsoqq2gSiRPrCJpblrK3Vvb22114QXSA)).

Could you provide a reproducible demo or repository?

**Comment by mattxwang** [comment]
Tagging myself in since I wrote the rule 😓 

(if there's a reproducible demo, happy to submit a patch to fix)

**Comment by luisdeka** [comment]
I use lower version "^14.10.x" but i have "stylelint-config-recommended-vue": "^1.4.0" and it uses higher version "15.3.0", i supposed the same problema.

All my *.vue files with <style lang="postcss"> throw this error at first line

**Comment by kaivin** [comment]
I think this problem will occur in any version, because my team is developing the same project, it is the same version, just because I have downloaded the dependency package a long time ago, my colleague is a recently cloned project and downloaded the project dependency package, he will have problems when submitting the project, I have no problem here, I have done testing, when I delete the node_modules and re-download the dependency package, when the code commits verification, this error appears

**Comment by EddyWorkHard** [comment]
> 
i am using the version 14.12.0 ,our team all appear this problem,we have to close stylelint ,hope this problem can fix asap.

**Comment by gen2wind** [comment]
I have the same @woldtwerk  is having, I even add stylint to git-commit hooks and now it is given me issue, I had to disable the commit hook relating to stylelint for this, I really hope there is a fix for this as soon as possible.

Thanks

**Comment by woldtwerk** [comment]
> Tagging myself in since I wrote the rule 😓 
> 
> (if there's a reproducible demo, happy to submit a patch to fix)

I'll try to add one asap

**Comment by mattxwang** [comment]
Ok, I think I'm understanding some of these comments more; if your issue is something of the form:

> Unknown rule selector-anb-no-unmatchable

Then I think it's highly likely that you're not on the latest version of Stylelint yet (`15.3.0` is the first --- and so far only --- version that has the rule). 

It sounds like some of the *other* configurations you're using, such as ones related to Vue, have not yet updated their peer dependency on Stylelint to the latest version (or it seems, the latest major version).

So, for @EddyWorkHard, @luisdeka, and @kaivin --- could you first make sure that the version of Stylelint you're using is exactly `15.3.0`?

---

@EddyWorkHard said:

> i am using the version 14.12.0 ,our team all appear this problem,we have to close stylelint ,hope this problem can fix asap.

It sounds like you have a config that relies on Stylelint `15.3` since it enables the rule. You should either:

1. upgrade Stylelint to v15.3, 
2. or, downgrade whatever config you updated recently until you're ready to upgrade Stylelint


---

@luisdeka said:

> I use lower version "^14.10.x" but i have "stylelint-config-recommended-vue": "^1.4.0" that uses "stylelint-config-recommended" and it uses higher version "15.3.0", i supposed is close or the same problem.
> 
> UPDATE: go to last version of stylelint and remove "stylelint-config-prettier" resolves the problem, but it will force people to update and remove stylelint prettier

As of Stylelint v15, [all stylistic rules have been deprecated](https://stylelint.io/migration-guide/to-15/#deprecated-stylistic-rules), and they are no longer enabled in `stylelint-config-recommended`. Those were the rules that conflicted with prettier; so, on v15, `stylelint-config-prettier` is not needed. You can verify this [on their documentation](https://github.com/prettier/stylelint-config-prettier):

> **Note**  
> As of Stylelint v15 [all style-related rules have been deprecated](https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules). If you are using v15 or higher and are not making use of these deprecated rules, [this plugin is no longer necessary](https://stylelint.io/migration-guide/to-15#:~:text=Additionally%2C%20you%20may%20no%20longer%20need%20to%20extend%20Prettier%27s%20Stylelint%20config).

Hope this helps!

**Comment by martinkruusement** [comment]
For me, this was fixed by restarting and clearing editor cache.
I believe Sublime Text was still using the old version through it's LSP plugin.

Removed `~/Library/Caches/Sublime Text` and the error is gone.

Config confirmed to work:

```json
{
    "extends":
    [
        "stylelint-config-recommended",
        "stylelint-config-standard",
        "stylelint-config-recommended-vue"
    ],
    "plugins":
    [
        "stylelint-order"
    ],
    "rules": { ... }, // should be irrelevant in this case, no changes made
    "files":
    [
        "*.vue",
        "**/*.vue",
        "*.css",
        "**/*.css"
    ]
}
```
Package.json > devDependencies:
```json
    "stylelint": "^15.3.0",
    "stylelint-config-recommended": "^11.0.0",
    "stylelint-config-recommended-vue": "^1.4.0",
    "stylelint-config-standard": "^25.0.0",
    "stylelint-order": "^5.0.0",
```

**Comment by gyhyfj** [comment]
same problem

**Comment by teinett** [comment]
Is it possible, that there is a misprint in the name of the rule?

Now: `selector-anb-no-unmatchable`
Expected: `selector-and-no-unmatchable` = second word should be "and"

**Comment by martinkruusement** [comment]
@teinett `anb` stands for An+B as in
`a:nth-child(0n+0) {}`

**Comment by hilyin** [comment]
I am getting this error on a brand new project created today.

New project creation showing error:

```
⇒  yarn create nuxt-app
yarn create v1.22.19
warning package.json: No license field
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Installed "create-nuxt-app@5.0.0" with binaries:
      - create-nuxt-app

create-nuxt-app v5.0.0
✨  Generating Nuxt.js project in .
? Project name: temp
? Programming language: TypeScript
? Package manager: Yarn
? UI framework: Bootstrap Vue
? Template engine: HTML
? Nuxt.js modules: (Press <space> to select, <a> to toggle all, <i> to invert selection)
? Linting tools: ESLint, Prettier, Lint staged files, StyleLint, Commitlint
? Testing framework: Jest
? Rendering mode: Universal (SSR / SSG)
? Deployment target: Server (Node.js hosting)
? Development tools: (Press <space> to select, <a> to toggle all, <i> to invert selection)
? Continuous integration: None
? Version control system: Git
yarn run v1.22.19
warning ../../package.json: No license field
$ prettier --write --list-different . && yarn lint:js --fix && yarn lint:style --fix
.eslintrc.js
commitlint.config.js
components/NuxtLogo.vue
components/Tutorial.vue
jest.config.js
nuxt.config.js
package.json
pages/index.vue
README.md
stylelint.config.js
tsconfig.json
warning ../../package.json: No license field
$ eslint --ext ".js,.ts,.vue" --ignore-path .gitignore . --fix
warning ../../package.json: No license field
$ stylelint "**/*.{css,scss,sass,html,vue}" --ignore-path .gitignore --fix

components/NuxtLogo.vue
 16:1  ✖  Unknown rule selector-anb-no-unmatchable  selector-anb-no-unmatchable

1 problem (1 error, 0 warnings)

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

🎉  Successfully created project temp

  To get started:

	yarn dev

  To build & start for production:

	yarn build
	yarn start

  To test:

	yarn test


  For TypeScript users.

  See : https://typescript.nuxtjs.org/cookbook/components/
✨  Done in 212.22s.
```

**Comment by mattxwang** [comment]
Hi @hilyin, please see [my reply above](https://github.com/stylelint/stylelint/issues/6719#issuecomment-1481705855) -- you are running into the same issue. In your case, it seems like a downstream dependency on Stylelint. I would suggest either reverting the change, raising an issue in their repository, or forcing the latest Stylelint version.

**Comment by mattxwang** [comment]
Circling back to this issue, it seems like all the cases are related to downstream consumers of Stylelint not properly updating their dependency chain. With our next minor release out, I'll close this issue for now; however, if others are encountering different issues, please comment or open a new issue - we'd love to help you out!

**Comment by tarreislam** [comment]
I solved my issue by going to stylelint 15.3 and removing "stylelint-config-prettier" since its deprecated since stylelint 15. problem solvedo
